### ISSUE TITLE
Search page results for un-downloaded movies are inaccessible 

### ISSUE THREAD

**Issue by coop4c** [Radarr/Radarr#9539]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

When navigating the search results with a screen reader the list reads as "unlabled button 0" this is because the button to click into the movie does not have an aria-label so screen readers do their best to identify it.

Example rendered html on page for such a button
```html
<button type="button" class="AddNewMovieSearchResult-underlay-EQMwD Link-link-RInnp"></button>
```

### Expected Behavior

The button should have a rendered aria-label making the button accessible to users who use screen readers. I have included an example below of a simple aria-label that when using a screen reader instead of reading "unlabeled button 0" reads "Open Stalin's James Bond 2017 Button".

```html
<button type="button" class="AddNewMovieSearchResult-underlay-EQMwD Link-link-RInnp" aria-label="Open Stalin's James Bond(2017)"></button>
```

### Steps To Reproduce

1. Install Radarr 
2. Access the homepage and search for movie (I used "James Bond" as my example)
3. Browse the results with a screen reader 

### Environment

```markdown
- OS: UnRaid 6.12
- Radarr: 5.2.6.8376
- Docker Install: Yes
- Using Reverse Proxy: No
- Browser: Chrome 120.0.6099.130
- Database: Sqlite 3.40.1
```

### What branch are you running?

Master

### Trace Logs? **Not Optional**

![image](https://github.com/Radarr/Radarr/assets/155028427/c39be2c0-c437-41d3-b5b7-744f16af40ee)

I have included trace logs even though they are not super relevant here. These logs should show me preforming an example search after preforming the search I browsed through the UI through a screen reader. [radarr.trace.txt](https://github.com/Radarr/Radarr/files/13813415/radarr.trace.txt)

### Trace Logs have been provided as applicable. Reports will be closed if the required logs are not provided.

- [X] I have read and followed the steps in the wiki link above and provided the required trace logs - the logs contain `trace` - that are relevant and show this issue.
